### ISSUE TITLE
MariaDB 10.6+ binaries packaging

### ISSUE THREAD

**Issue by cobbce** [MariaDB4j/MariaDB4j#344]
Wanting to get my hands on that sweet sweet Insert... Returning capability and save myself an extra db round trip.

Very excited to discover this project, I'd left H2 behind long since as they didn't keep up with mysql/mariadb syntax evolution, forcing me to write conditionals in production code to change up my syntax, which kind of defeats the purpose of writing integration tests (because the production code path is no longer exercised).

Basically this project is wonderful, keep up the good work!

**Comment by vorburger** [comment]
Thank You! Your contribution to add 10.5 would be very very welcome. There is doc here https://github.com/vorburger/MariaDB4j/blob/master/README.md#mariadb-database-jars-and-version-upgrades how to do it. Just ask if not clear, and we'll try to help.

**Comment by benweizhu** [comment]
We need the 10.5+ too.

**Comment by luigiberrettini** [comment]
Hi @vorburger I read the README and saw that binaries are not supposed to be added to this repo anymore.

Besides the changes to POMs like [the ones you did](https://github.com/vorburger/MariaDB4j/commit/7fb1bd9c625f0867a425a23ba554e2095f919cda) recently, I am not sure where contributors need to submit a PR with the binaries, for them to be packaged as JAR and pushed to Maven central.

**Comment by vorburger** [comment]
>  I am not sure where contributors need to submit a PR with the binaries, for them to be packaged as JAR and pushed to Maven central.

Here in this project, but you need to raise a PR like #235 - without binaries, but just downloading from a safe well known URL.

> read the README and saw that binaries are not supposed to be added to this repo anymore.

I was trying to find where that's written clearly, and myself can't see it now.. where did you see this? I've tried to point someone else to it over in #363. If you want to contribute to the project, a PR updating the README to make it clearer how to do this for contributors in the future would be very welcome.

**Comment by luigiberrettini** [comment]
When talking about binaries I was referring to the DB binaries and as per some issues (not the README) and what you said above they should no more be committed to git and therefore they will be no more available in this repo.

The new way to provide MariaDB updated versions is, per my understanding, creating folders and pom.xml inside them as for the 10.3 version: binaries will be downloaded from known locations on build and used to create JARs but not be anymore stored in this repo.

I would suggest keeping one folder per major.minor version and trying to update the patch version as much as possible.

**Comment by luigiberrettini** [comment]
My understanding is that Travis CI is used to build the DB JARs and the MariaDB4j JAR.

It is not clear to me how new MariaDB4j versions are tested against each current/new DB version and how JARs are uploaded to Maven central: I do not see build steps doing so.

**Comment by vorburger** [comment]
> It is not clear to me how new MariaDB4j versions are tested against each current/new DB version 

They are not, only the last one. If you need it to be for each, please contribute an enhancement to do so.

> and how JARs are uploaded to Maven central: I do not see build steps doing so.

The actual release step itself is something I need to do, after people contribute the required PRs.

**Comment by vorburger** [comment]
@luigiberrettini it looks like @bgalek over in #363 is interested in the same. You two should talk... :smile_cat:

**Comment by luigiberrettini** [comment]
> They are not, only the last one. If you need it to be for each, please contribute an enhancement to do so.

Since the repo has older versions and on Maven central we can download them it sounds reasonable to see the library as something more or less separate from MariaDB: its goal is starting a DB and ideally the latest version of each series should be launchable (apart of probably the ones that reached [end of life](https://endoflife.date/mariadb)).

It would be nice to have the build script allow not only build and test but also release: by using secrets encrypted with your key you are the only one that can perform the release just pressing a button or adding a tag.

I just submitted a PR with the latest version of MariaDB 10.3.x and I have the ones related to MariaDB 10.4.x and 10.5.x ready to submit.

We can then work on understanding what are future plans in terms of versions to support, how to reduce duplication when a version is added and if it is possible to dynamically support versions (without PRs and downloading them at usage time so that we can potentially support many versions easily).

**Comment by vorburger** [comment]
#560 is a new take on this project's old "package newer binaries" topic... with that, one could easily use latest "local" binaries.

**Comment by vorburger** [comment]
@cobbce @luigiberrettini FYI in #771 @TheKnowles has contributed (and it got merged as part of #772) a newer version MariaDB binaries. With that, I am therefore now closing this 2 years old issue.
